search-app/src/cmr/search/api/concepts_search.clj

```diff
@@ -366,13 +367,36 @@
   ;; no errors, return 204
   {:status 204})
 
+(defn pluralize-concept
+  "Return the plural form of a concept"
+  [concept]
+  (let [concept-name (name concept) last-letter (subs concept-name (- (count concept-name) 1))]
+    (if (= "y" last-letter)
+      (keyword (str (subs concept-name 0 (- (count concept-name) 1)) "ies"))
+      (keyword (str concept-name "s")))))
+
+(defn format-regex
+  "Adds the necessary string to each concept for a regex"
+  [concept] (str "(?" concept ")"))
+
+(def get-generics
+  "Retrieve the generics which were dynamically loaded, drop the keyword :generic"
+  (mapv pluralize-concept (drop 1 (keys concepts/generic-concept-types->concept-prefix))))
+
+(def join-generic-concepts
+  "Combines the generic concepts to be a single string"
+  (clojure.string/join "|" (mapv format-regex get-generics)))
+
+(def routes-regex
+  "Appends the generic concepts dynamically loaded to the non-generic concepts to match possible routes"
+  (re-pattern (str "(?:(?:granules)|(?:collections)|(?:variables)|(?:subscriptions)|(?:tools)|(?:services)|" join-generic-concepts ")(?:\\..+)?")))
```

Comment:
why ?:
Not saying your wrong, regex is a bit of magic, but is it required?

Reply:
I'm retracting my question, it looks like you were using an already existing regex.